safehttp/plugins/csp/csp_test.go

```diff
@@ -0,0 +1,253 @@
+			wantNonce:            "KSkpKSkpKSkpKSkpKSkpKSkpKSk=",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+
+			tt.interceptor.Before(rr.ResponseWriter, req)
+
+			h := rr.Header()
+			if diff := cmp.Diff(tt.wantEnforcPolicy, h.Values("Content-Security-Policy"), cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy\") mismatch (-want +got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tt.wantReportOnlyPolicy, h.Values("Content-Security-Policy-Report-Only"), cmpopts.EquateEmpty()); diff != "" {
+				t.Errorf("h.Values(\"Content-Security-Policy-Report-Only\") mismatch (-want +got):\n%s", diff)
+			}
+
+			v := req.Context().Value(ctxKey{})
+			if v == nil {
+				t.Fatalf("req.Context().Value(ctxKey{}) got: nil want: %q", tt.wantNonce)
+			}
+			if got := v.(string); got != tt.wantNonce {
+				t.Errorf("v.(string) got: %q want: %q", got, tt.wantNonce)
+			}
+		})
+	}
+}
+
+func TestAlreadyClaimed(t *testing.T) {
+	headers := []string{
+		"Content-Security-Policy",
+		"Content-Security-Policy-Report-Only",
+	}
+
+	for _, h := range headers {
+		t.Run(h, func(t *testing.T) {
+			rr := safehttptest.NewResponseRecorder()
+			if _, err := rr.ResponseWriter.Header().Claim(h); err != nil {
+				t.Fatalf("rr.ResponseWriter.Header().Claim(h) got err: %v want: nil", err)
+			}
+			req := safehttptest.NewRequest(safehttp.MethodGet, "/", nil)
+
+			it := Interceptor{}
+			it.Before(rr.ResponseWriter, req)
+
+			if got, want := rr.Status(), int(safehttp.StatusInternalServerError); got != want {
+				t.Errorf("rr.Status() got: %v want: %v", got, want)
+			}
+
+			wantHeaders := map[string][]string{
+				"Content-Type":           {"text/plain; charset=utf-8"},
+				"X-Content-Type-Options": {"nosniff"},
+			}
+			if diff := cmp.Diff(wantHeaders, map[string][]string(rr.Header())); diff != "" {
+				t.Errorf("rr.Header() mismatch (-want +got):\n%s", diff)
+			}
+
+			if got, want := rr.Body(), "Internal Server Error\n"; got != want {
+				t.Errorf("rr.Body() got: %q want: %q", got, want)
+			}
+		})
+	}
+}
+
+type errorReader struct{}
+
+func (errorReader) Read(b []byte) (int, error) {
+	return 0, errors.New("bad")
+}
+
+func TestPanicWhileGeneratingNonce(t *testing.T) {
+	randReader = errorReader{}
+	defer func() {
+		if r := recover(); r == nil {
+			t.Error("generateNonce() expected panic")
+		}
+	}()
+	generateNonce()
+}
+
+func TestValidNonce(t *testing.T) {
+	ctx := context.WithValue(context.Background(), ctxKey{}, "nonce")
+	n, err := Nonce(ctx)
+	if err != nil {
+		t.Errorf("Nonce(ctx) got err: %v want: nil", err)
+	}
+
+	if want := "nonce"; n != want {
+		t.Errorf("Nonce(ctx) got nonce: %v want: %v", n, want)
+	}
+}
+
+func TestNonceEmptyContext(t *testing.T) {
+	ctx := context.Background()
+	n, err := Nonce(ctx)
+	if err == nil {
+		t.Error("Nonce(ctx) got err: nil want: error")
+	}
+
```

Comment:
nit: remove empty line

Reply:
I like to separate each different check in a test with empty lines. This way you can easily see that here comes a new check. If I should remove this empty line then the empty line in `TestValidNonce` should also be removed for consistency. What do you think?